File: jetpackmvvm/src/main/java/com/zxn/mvvm/model/IApiFactory.java
package com.zxn.mvvm.model;


import androidx.annotation.IntDef;

import java.io.File;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Created by zxn on 2019/4/30.
 */
public interface IApiFactory {

    /**
     * 默认的缓存大小.
     */
    long DEFAULT_MAX_CACHE_SIZE = 10 * 1024 * 1024;

    int MODE_ON_LINE = 0;//线上
    int MODE_ON_TEST = 1;//测试
    int MODE_ON_GRAY = 2;//灰度


    @IntDef({MODE_ON_LINE, MODE_ON_TEST, MODE_ON_GRAY})
    @Retention(RetentionPolicy.SOURCE)
    @interface ModeDef {
    }

    /**
     * 切换开发模式,
     *
     * @param mode
     */
    void setMode(@ModeDef int mode);

    /**
     * 设置缓存的路径.
     *
     * @return File
     */
    File cacheFile();

    /**
     * 设置缓存的最大值.
     *
     * @return long, 存的最大值
     */
    long maxCacheSize();

    /**
     * 控制日志输出.
     *
     * @return true的时候打印日志.
     */
    boolean isShowLog();


}

File: jetpackmvvm/src/main/java/com/zxn/mvvm/model/IApiFactory.kt
package com.zxn.mvvm.model

import androidx.annotation.IntDef
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * Created by zxn on 2019/4/30.
 */
interface IApiFactory {
    @IntDef(MODE_ON_LINE, MODE_ON_TEST, MODE_ON_GRAY)
    @Retention(RetentionPolicy.SOURCE)
    annotation class ModeDef

    /**
     * 切换开发模式,
     *
     * @param mode
     */
    fun setMode(@ModeDef mode: Int)

    companion object {
        /**
         * 默认的缓存大小.
         */
        const val DEFAULT_MAX_CACHE_SIZE = 10 * 1024 * 1024.toLong()
        const val MODE_ON_LINE = 0 //线上
        const val MODE_ON_TEST = 1 //测试
        const val MODE_ON_GRAY = 2 //灰度
    }
}